MCMC/simulation.py

```diff
@@ -53,18 +59,46 @@ def trial_move(self):
                 generate a random number p'
                 if p' < p : Accept else reject
         """
-        return NotImplementedError
+        _system = np.copy(self.system)
+        # Select random occupied site
+        non_zero_indices = list(np.nonzero(self.system))
+        random.shuffle(non_zero_indices)
+        site1 = non_zero_indices[0]
+        site1_val = _system[site1]
+        # Select random site from all possible sites
+        all_indices = list(self.grid_indices) 
```

Comment:
Do we wanna exclude `site1` from this list of potential candidates?

Reply:
I thought about that, but not sure if that is considered a valid move and we want to keep it as a possible out come (i.e. nothing changes during this step).